src/blosc2/proxy.py

```diff
@@ -129,6 +129,25 @@ def fetch(self, item=None):
         -------
         out: :ref:`NDArray` or :ref:`SChunk`
             The local container used to cache the already requested data.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> import blosc2
+        >>> data = np.arange(20).reshape(10, 2)
+        >>> ndarray = blosc2.asarray(data)
+        >>> proxy = blosc2.Proxy(ndarray)
+        >>> full_data = proxy.fetch()
+        >>> f"Full data cache: {full_data[:]}"
+        Full data cache:
+            [[ 0  1][ 2  3][ 4  5]
+            [ 6  7][ 8  9][10 11]
+            [12 13][14 15][16 17]
+            [18 19]]
+        >>> slice_data = proxy[0:2, :]
+        >>> f"Slice data cache: {slice_data}"
```

Comment:
Here it makes little sense to show the slice, as the full array has been fetched already.  Better put this above, as already suggested.